packages/@aws-cdk/aws-rds/lib/instance-engine.ts

```diff
@@ -0,0 +1,526 @@
+ * but there might be fields added to it in the future.
+ */
+export interface InstanceEngineConfig {
+}
+
+/**
+ * Interface representing a database instance (as opposed to cluster) engine.
+ */
+export interface IInstanceEngine {
+  /** The type of the engine, for example "mysql". */
+  readonly engineType: string;
+
+  /**
+   * The exact version of the engine that is used,
+   * for example "5.1.42".
+   *
+   * @default - use the default version for this engine type
+   */
+  readonly engineVersion?: string;
+
+  /** The application used by this engine to perform rotation for a single-user scenario. */
+  readonly singleUserRotationApplication: secretsmanager.SecretRotationApplication;
+
+  /** The application used by this engine to perform rotation for a multi-user scenario. */
+  readonly multiUserRotationApplication: secretsmanager.SecretRotationApplication;
+
+  /**
+   * The default instanceType to use with this engine.
+   */
+  readonly defaultInstanceType: ec2.InstanceType;
+
+  /**
+   * Method called when the engine is used to create a new instance.
+   */
+  bindToInstance(scope: core.Construct, options: InstanceEngineBindOptions): InstanceEngineConfig;
+}
+
+interface InstanceEngineBaseProps {
+  readonly engineType: string;
+  readonly singleUserRotationApplication: secretsmanager.SecretRotationApplication;
+  readonly multiUserRotationApplication: secretsmanager.SecretRotationApplication;
+  readonly parameterGroupFamilies?: ParameterGroupFamilyMapping[];
+  readonly defaultInstanceType?: ec2.InstanceType;
+  readonly engineVersion?: string;
+}
+
+abstract class InstanceEngineBase implements IInstanceEngine {
+  public readonly engineType: string;
+  public readonly engineVersion?: string;
+  public readonly singleUserRotationApplication: secretsmanager.SecretRotationApplication;
+  public readonly multiUserRotationApplication: secretsmanager.SecretRotationApplication;
+  public readonly defaultInstanceType: ec2.InstanceType;
+
+  constructor(props: InstanceEngineBaseProps) {
+    this.engineType = props.engineType;
+    this.singleUserRotationApplication = props.singleUserRotationApplication;
+    this.multiUserRotationApplication = props.multiUserRotationApplication;
+    this.defaultInstanceType = props.defaultInstanceType ?? ec2.InstanceType.of(ec2.InstanceClass.M5, ec2.InstanceSize.LARGE);
+    this.engineVersion = props.engineVersion;
+  }
+
+  public bindToInstance(_scope: core.Construct, _options: InstanceEngineBindOptions): InstanceEngineConfig {
+    return {
+      defaultInstanceType: this.defaultInstanceType,
+    };
+  }
+}
+
+abstract class NonSqlServerInstanceEngine extends InstanceEngineBase {
+  public bindToInstance(scope: core.Construct, options: InstanceEngineBindOptions): InstanceEngineConfig {
+    if (options.timezone) {
+      throw new Error(`timezone property can be configured only for Microsoft SQL Server, not ${this.engineType}`);
+    }
+    return super.bindToInstance(scope, options);
+  }
+}
```

Comment:
I did this change, but, in my opinion, it's strictly worse than the proposed solution in this revision.

If the code for `bindToInstance()` did some interesting logic besides just validating `timezone`, you would have to duplicate it in `SqlServerInstanceEngine` (you can't call `super` there anymore, because it will do the incorrect validation).